src/components/Modals/NetworkSwitchModal/NetworkSwitchModal.js

```diff
@@ -0,0 +1,132 @@
+import React, { useCallback } from 'react'
+import PropTypes from 'prop-types'
+
+import { GU, Modal, useTheme, useViewport, textStyle, Button } from '@aragon/ui'
+import styled from 'styled-components'
+
+import { useWallet } from '../../../contexts/wallet'
+import { getNetworkShortName, getNetworkFullName } from '../../../util/network'
+
+NetworkSwitchModal.propTypes = {
+  visible: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+}
+
+export function NetworkSwitchModal({ onClose, visible }) {
+  const theme = useTheme()
+  const { below } = useViewport()
+  const smallMode = below('medium')
+  const { networkType } = useWallet()
+  const networkName = getNetworkFullName(networkType)
+
+  const modalWidth = useCallback(
+    ({ width }) => Math.min(72 * GU, width - 4 * GU),
+    []
+  )
+
+  return (
+    <Modal visible={visible} width={modalWidth} onClose={onClose}>
+      <Content smallMode={smallMode}>
+        <Header>
+          <Title>Select a Network</Title>
+          <Subtitle color={theme.contentSecondary}>
+            You are currently connected to the <b>{networkName}</b> network
+          </Subtitle>
+        </Header>
+        <Body>
+          <div>
+            <NetworkTitle>Mainnets</NetworkTitle>
+            <ButtonsRow networkNames={mainNetworksTypes} onClose={onClose} />
+          </div>
+          <div>
+            <NetworkTitle>Testnets</NetworkTitle>
+            <ButtonsRow networkNames={testNetworksTypes} onClose={onClose} />
+          </div>
+        </Body>
+      </Content>
+    </Modal>
+  )
+}
+
+// add new networks here
+const mainNetworksTypes = ['main', 'matic']
```

Comment:
Can we put a property in the networkConfig https://github.com/aragon/client/blob/master/src/network-config.js, to filter this information out? Try to keep all config in 1 spot in case we need to add more chain later...

Reply:
Is there still more PR for this task? I thought the modal should only display the networks where the DAO exists... not hardcoding.. So, if the DAO does not exist anywhere, it would display DAO not found error...

Reply:
Let me give you some context here, just so we're on the same page. The array(s) here says: all networks that are in here should be displayed as buttons on the modal, so that the user can switch to there. (note, there are more chains in the network config object than we actually use)

Now, I don't think that's a network property in and of itself 😅. I think it's a property of the modal, which is why I put here. Just to be clear: it is very much possible to put this info into the network config object. But then you'd have a boolean property like: shouldAppearOnModal. If you think that's the way to go I'll add it 👍🏼

Reply:
No, the task is done, otherwise I wouldn't have requested a PR. You can find more information about it on the respective ticket on Jira 👍🏼